### DISCUSSION TITLE
[Bug]: Value of form field tied to a checkbox group not consistent

### DISCUSSION THREAD

**Discussion by txr59v** [react-hook-form/react-hook-form#6116] [Q&A]
### Version Number

7.12.1, 6.15.8

### Codesandbox

https://codesandbox.io/s/rhf-multiple-checkboxes-validation-forked-09s6i?file=/src/index.js

### Steps to reproduce

# On 7.12.1

## 1 option behavior
1. Create a checkbox group with 1 checkbox
2. Set checkbox group value to a default empty array: `[]`
3. Inspect the value of the form after the last render of the component
4. The form's checkbox group value will be the empty array and the checkbox will be checked
5. Uncheck the single checkbox
6. The form's checkbox group value will be `false`
7. Check the single checkbox
8. The form's checkbox group value will be the value of the checkbox

## 2+ option behavior
1. Create a checkbox group with 2+ checkboxes
2. Set checkbox group value to a default empty array: `[]`
3. Inspect the value of the form after the last render of the component
4. The form's checkbox group value will be the empty array and no checkboxes will be checked
5. Check a checkbox
6. The form's checkbox group value will be an array with 1 item which is the value of the checkbox
7. Uncheck the checkbox
4. The form's checkbox group value will be the empty array

# On 6.15.8

## 1 option behavior
1. Create a checkbox group with 1 checkbox
2. Set checkbox group value to a default empty array: `[]`
3. Inspect the value of the form after the last render of the component
4. The form's checkbox group value will be the value of the checkbox and the checkbox will be checked
5. Uncheck the single checkbox
6. The form's checkbox group value will be `false`

### Expected behaviour

I would expect the behavior of 1 option and 2+ option to be identical so a dynamically rendered list of checkboxes can have the form value handled the same. My project currently uses 6.15.4 which is why I included 6.15.8 behavior but I noticed the inconsistency in 7 as well, just slightly different behavior for what the form is set to.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
V7 is expected behaviour, if this is not desired please build a controlled component and host your own local state

single checkbox 
false or value

multiple
[] or [xxx, xxx]
